modal/stub.py

```diff
@@ -236,10 +236,11 @@ def _add_object(self, tag, obj):
         self._indexed_objects[tag] = obj
 
     def __getitem__(self, tag: str):
-        # Deprecated? Note: this is currently the only way to refer to lifecycled methods on the stub, since they have . in the tag
+        deprecation_warning((2024, 3, 23), "stub[...] is deprecated!")
```

Comment:
Nit: I had trouble parsing this for a second, maybe elaborating a bit would help, e.g.

```
"Stub subscription (`stub[...]`) is deprecated"
```

(though "subscription" is a fairly arcane term, the slightly less accurate "indexing" may be more familiar).

Also do we want to push people towards setattr/getattr or no because we also plan on deprecating those? I don't think deprecation churn is great UX but neither is a deprecation message that doesn't suggest a remediation.

Reply:
There really isn't much of a reason to use either setattr or setitem at this point – people should be able to remove it. The only exception would be `.new()` methods which are themselves deprecated.

I can make that clear in the error message!

Reply:
Yeah — saying something like "It is no longer necessary as objects can be defined in global scope" seems helpful!